packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1904,7 +1904,16 @@ abstract class BuildContext {
   /// called, whenever changes occur relating to that widget until the next time
   /// the widget or one of its ancestors is moved (for example, because an
   /// ancestor is added or removed).
-  InheritedWidget inheritFromWidgetOfExactType(Type targetType);
+  ///
+  /// The [aspect] parameter is only used when [targetType] is a subclass
+  /// of [InheritedModel]. It specifies what "aspect" of the inherited model
+  /// widget this context depends on.
+  ///
+  /// If the [target] parameter is specified its [widget] must be a [targetType]
+  /// ancestor of this BuildContext's widget. The target is used instead of
+  /// the first [targetType] widget ancestor. This parameter is used by
+  /// [InheritedModel.inheritFrom]. It's typically not specified directly.
+  InheritedWidget inheritFromWidgetOfExactType(Type targetType, { Object aspect, InheritedElement target });
```

Comment:
can we refactor this so instead of one method that does the walk if target is null and then depends on target, we have two methods, one for doing the walk and one for actually setting up the inheritance? i think that would be cleaner. It's weird to say "inheritFromWidgetOfExactType" and then provide the target (at which point the type isn't even relevant?).

Reply:
I agree that it would cleaner to just a separate method for this case.

The reason I'd done things this way was to ensure that the checking done by StatefulElement.inheritFromWidgetOfExactType() would happen, even if the target was specified. The big error message there would be a little harder to understand if it was generated by a new method rather than the one the user called directly.

I'll take a crack at refactoring things and dealing with StatefulElement's error reporting.